examples/cis2-wccd-upgradable/src/lib.rs

```diff
@@ -0,0 +1,3537 @@
+        EntrypointName::new_unchecked("initialize"),
+        Amount::zero(),
+    )?;
+
+    let implementation_address = host.state().implementation_address;
+
+    let parameter_bytes = to_bytes(&InitializeImplementationParams {
+        proxy_address: ctx.self_address(),
+        state_address: host.state().state_address,
+    });
+
+    host.invoke_contract_raw(
+        &implementation_address,
+        Parameter(&parameter_bytes),
+        EntrypointName::new_unchecked("initialize"),
+        Amount::zero(),
+    )?;
+
+    // Log a `Mint` event for the single token id with 0 amount. Minting an amount
+    // of 0 is used to signal that a new token was deployed. Invoking the mint event
+    // with 0 amount in the `wrap` function is impossible.
+    logger.log(&Cis2Event::Mint(MintEvent {
+        token_id: TOKEN_ID_WCCD,
+        amount:   ContractTokenAmount::from(0u64),
+        owner:    ctx.sender(),
+    }))?;
+
+    // Log an event including the metadata for this token.
+    logger.log(&Cis2Event::TokenMetadata::<_, ContractTokenAmount>(TokenMetadataEvent {
+        token_id:     TOKEN_ID_WCCD,
+        metadata_url: MetadataUrl {
+            url:  String::from(TOKEN_METADATA_URL),
+            hash: None,
+        },
+    }))?;
+
+    // Log a new implementation event.
+    logger.log(&WccdEvent::NewImplementation(NewImplementationEvent {
+        new_implementation: implementation_address,
+    }))?;
+
+    // Log a new admin event.
+    logger.log(&WccdEvent::NewAdmin(NewAdminEvent {
+        new_admin: host.state().admin,
+    }))?;
+
+    Ok(())
+}
+
+/// The fallback method, which redirects the invocations to the implementation.
+#[receive(contract = "CIS2-wCCD-Proxy", fallback, mutable, payable)]
+fn receive_fallback<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<StateProxy, StateApiType = S>,
+    amount: Amount,
+) -> ReceiveResult<RawReturnValue> {
+    let entrypoint = ctx.named_entrypoint();
+    let implementation = host.state().implementation_address;
+
+    let mut parameter_buffer = vec![0; ctx.parameter_cursor().size() as usize];
+    ctx.parameter_cursor().read_exact(&mut parameter_buffer)?;
+
+    // Appending the sender to the input parameters
+    let paramter_with_sender = ParamWithSender {
+        params: parameter_buffer,
+        sender: ctx.sender(),
+    };
+
+    let parameter_bytes = to_bytes(&paramter_with_sender);
+
+    // Forwarding the invoke unaltered to the implementation contract.
+    let mut return_value = host
+        .invoke_contract_raw(
+            &implementation,
+            Parameter(&parameter_bytes),
+            entrypoint.as_entrypoint_name(),
+            amount,
+        )
+        .map_err(|r| {
+            if let CallContractError::LogicReject {
+                reason,
+                ..
+            } = r
+            {
+                Reject::new(reason).unwrap_abort()
+            } else {
+                r.into()
+            }
```

Comment:
For now this is fine, but I don't think it will be necessary after https://github.com/Concordium/concordium-rust-smart-contracts/pull/145 gets merged